### DISCUSSION TITLE
Speed up Earth file loading

### DISCUSSION THREAD

**Discussion by devil-darkside** [pelicanmapping/osgearth#2446] [Support Q&A] [answered]
Is there a way to speed up the loading of an earth file?
I have built the overview for my imagery and then I have generated the vrt with gdal for both imagery and elevation file.
The problem is that even if I tried to optimize my data the earth file took a lot of time for loading.
Any suggestions?

**Comment by gwaldron** [comment]
There are several tips on preprocessing and optimizing your data [here](https://docs.osgearth.org/en/latest/data.html).

**Comment by devil-darkside** [comment]
I have already optimized my geotiff by adding overviews (last overview is 128x128 pixel) and generated the spatial index for my shapefiles.
The process is always slow, I have a VRT of 5000 geotiffs at high resolution.

**Reply by jasonbeverage** [reply, marked as answer]
5000 geotiffs at high resolution can be quite difficult to load using the GDAL driver even if they have overviews.  You should consider creating an mbtiles or a TMS dataset out of the data using [osgearth_conv](https://docs.osgearth.org/en/latest/osgearth_conv.html) (or any other tool) or put your data behind a server that caches tiles so they aren't generated each time.

**Reply by devil-darkside** [reply]
The startup is quite long with all these file.
I'll try to create several VRT to undestand if I can speed up some processes.
I'm also using the cache option to do not generate each time the tiles and it works fine. 
I'll try to generate a tms and/or mbtiles and check performances to see if this could be a possibile solution.
